### PR TITLE
Openjdk new subport and Update openjdk

### PR DESCRIPTION
**Pull request by usersxx**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G65 x86_64
 
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: java/openjdk/Portfile [hunk 1]
```diff
@@ -101,7 +104,7 @@ subport openjdk7-zulu {
 }
 
 subport openjdk8 {
-    version      8u302
+    version      8u322
```
Comment (breun) [inline java/openjdk/Portfile:107]: I don't see why this version needs to be bumped. I get that it's tempting to let it match the latest OpenJDK version, but bumping this version will just cause an unnecessary update for everyone who has this empty port installed without any functional benefit.

File: java/openjdk/Portfile [hunk 2]
```diff
@@ -208,7 +211,7 @@ subport openjdk8-openj9-large-heap {
 }
 
 subport openjdk11 {
-    version      11.0.12
+    version      11.0.14.1
```
Comment (breun) [inline java/openjdk/Portfile:214]: It's not really necessary to bump this version either.

File: java/openjdk/Portfile [hunk 3]
```diff
@@ -273,6 +276,24 @@ subport openjdk11-openj9-large-heap {
     replaced_by  openjdk11-openj9
 }
 
+subport openjdk11-sap {
+    # https://github.com/SAP/SapMachine/releases/tag/
```
Comment (breun) [inline java/openjdk/Portfile:280]: There is no page at this URL. I suggest using https://github.com/SAP/SapMachine/releases/ instead.

File: java/openjdk/Portfile [hunk 4]
```diff
@@ -595,6 +616,30 @@ subport openjdk17-oracle {
     worksrcdir   jdk-${version}.jdk
 }
 
+subport openjdk17-sap {
+    # https://github.com/SAP/SapMachine/releases/tag/
```
Comment (breun) [inline java/openjdk/Portfile:620]: There is no page at this URL. I suggest using https://github.com/SAP/SapMachine/releases/ instead.

**Review by breun — CHANGES_REQUESTED** [review]

**Comment by usersxx** [comment]
@pmetzger you can merge this request if @breun agrees to

**Comment by breun** [comment]
I'm fine with adding these. Will you also be using these builds yourself, @usersxx? Because there are still quite a few other Java distributions that could be added (Amazon Coretto, Bellsoft Liberica, etc.), but I wonder if MacPorts users will actually use all of them.

**Comment by usersxx** [comment]
> I'm fine with adding these. Will you also be using these builds yourself, @usersxx? Because there are still quite a few other Java distributions that could be added (Amazon Coretto, Bellsoft Liberica, etc.), but I wonder if MacPorts users will actually use all of them. And if I need to maintain all of them myself...

There needs to be advantages of these builds or no one would install them.
Few benefits of these builds are (based on my experience):
1. Azul Zulu ( `openjdk*-zulu` ) : 
-  Modified Open Source components of OpenJDK
-  TCK tested
-  Long support
2. Eclipse Temurin ( `openjdk*-temurin` ) : 
-  Open Source components of OpenJDK
-  Least modifications
- Alternative Suggested OpenJDK boot JDK (Successor of AdoptOpenJDK)
3. IBM Semeru (`openjdk*-openj9` ) : 
- Uses Eclipse Openj9
4. Oracle GraalVM ( `openjdk*-graalvm` ) : 
-  An project of OpenJDK (https://openjdk.java.net/projects/graal/)
- Support multiple language with Java
5. Oracle OpenJDK (`openjdk*-oracle`) : 
-  Open Source version of Oracle Java SE
- Suggested for OpenJDK boot JDK (https://github.com/openjdk/jdk/blob/master/doc/building.md#boot-jdk-requirements)

**Comment by usersxx** [comment]
> I'm fine with adding these. Will you also be using these builds yourself, @usersxx? Because there are still quite a few other Java distributions that could be added (Amazon Coretto, Bellsoft Liberica, etc.), but I wonder if MacPorts users will actually use all of them. And if I need to maintain all of them myself...

There needs to be advantages of these builds or no one would install them.
Few benefits of these builds are (based on my experience):
1. Azul Zulu ( `openjdk*-zulu` ) : 
-  Modified Open Source components of OpenJDK
-  TCK tested
-  Long support
- https://www.azul.com/downloads/
- Built directly from OpenJDK source with replacement to almost all components
2. Eclipse Temurin ( `openjdk*-temurin` ) : 
-  Open Source components of OpenJDK
-  Least modifications
- TCK tested
- Alternative Suggested OpenJDK boot JDK (https://github.com/openjdk/jdk/blob/master/doc/building.md#boot-jdk-requirements) (Successor of AdoptOpenJDK)
- https://adoptium.net/
- Built directly from OpenJDK source with replacement to few components
3. IBM Semeru (`openjdk*-openj9` ) : 
- Uses Eclipse Openj9
- https://developer.ibm.com/languages/java/semeru-runtimes/
4. Oracle GraalVM ( `openjdk*-graalvm` ) : 
-  An project of OpenJDK (https://openjdk.java.net/projects/graal/)(Open Source)
- TCK tested
- Support multiple language including Java
- https://www.graalvm.org/
- Source (depreciated: https://hg.openjdk.java.net/graal | Moved to: https://github.com/oracle/graal) 
5. Oracle OpenJDK (`openjdk*-oracle`) : 
-  Open Source version of Oracle Java SE
- TCK tested
- Suggested for OpenJDK boot JDK (https://github.com/openjdk/jdk/blob/master/doc/building.md#boot-jdk-requirements)
- https://jdk.java.net
- Built directly from OpenJDK source (https://github.com/openjdk/jdk) (https://openjdk.java.net/install)
6. Sap SE SapMachine (`openjdk*-sap`) : 
- Light Weight JDK
- Open Source components of OpenJDK
- https://sap.github.io/SapMachine/
- Source (https://github.com/SAP/SapMachine)
7. Bellsoft Liberica:
- Portable
- TCK tested
- https://bell-sw.com/pages/downloads/
- Built directly from OpenJDK source with few modification and additions
8. Amazon Coretto:
- TCK tested
- https://aws.amazon.com/corretto/
- Source (https://github.com/corretto/corretto-jdk)

**Comment by usersxx** [comment]
I added them to build sapmachine sources

**Comment by usersxx** [comment]
I added them to build SapMachine from sources with SapMachine JDK itself

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
> I'm fine with adding these. Will you also be using these builds yourself, @usersxx? Because there are still quite a few other Java distributions that could be added (Amazon Coretto, Bellsoft Liberica, etc.), but I wonder if MacPorts users will actually use all of them. And if I need to maintain all of them myself...

Yes, the number of alternative OpenJDK distributions is almost dizzying, and not necessarily needed. (Unless there's user demand, of course.)

So let's merge this; we can always add more later.

**Pull request merged**